controllers/openstackdataplanerole_controller.go

```diff
@@ -157,9 +153,9 @@ func (r *OpenStackDataPlaneRoleReconciler) Reconcile(ctx context.Context, req ct
 	if len(instance.Spec.BaremetalSetTemplate.BaremetalHosts) > 0 {
 		ctrlResult, err := r.ReconcileBaremetalSet(ctx, instance, helper)
 		if err != nil {
-			return ctrlResult, err
-		} else if (ctrlResult != ctrl.Result{}) {
-			return ctrlResult, nil
```

Comment:
returning and checking ctrlResult as `nil` is not standard and confusing.  We could return just  a 'isReady' boolean from ReconcileBaremetalSet() rather than *ctrl.Result I guess. Also, below check can be simplified.
if (err != nil || ctrlResult != nil) {
    return ctrlResult, err}

Reply:
Done.